duckduckgo/iOS
# request desktop site

Task/Issue URL: https://app.asana.com/0/392891325557410/375790519910149
Tech Design URL: 
CC:

**Description**:

Add request desktop site functionality - mimicking our Android capability for now.

**Steps to test this PR**:

1. Go to a number of sites and request desktop - should see desktop version*
1. Request mobile - should return to mobile version
1. Check pixels
1. Open multiple tabs (each tab is independently desktop/mobile)

*Note that some sites are responsive and won't show any differences - this is a known limitation for now.  Some that support it are:

* facebook.com
* geartest.com

You can also check the user agent at by doing a search for user agent at DuckDuckGo.com

*Pixels*

Pixels are fired as follows:

* browsing Menu Opened = "mb"
* browsing Menu Refresh = "mb_rf"
* browsing Menu New Tab = "mb_tb"
* browsing Menu Add To Bookmarks = "mb_abk"
* browsing Menu Remove Bookmark = "mb_rbk"
* browsing Menu Toggle Browsing Mode = "mb_dm"
* browsing Menu Share = "mb_sh"
* browsing Menu Whitelist = "mb_wl"
* browsing Menu Report Broken Site = "mb_rb"
* browsing Menu Settings = "mb_st"

Go to Kibana and ensure each menu option fires the correct pixel.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
